tests/test_config/test_matcha_config.py

```diff
@@ -0,0 +1,243 @@
+"""Tests for Matcha Config module."""
+
+import json
+import os
+from typing import Dict, Iterator
+
+import pytest
+
+from matcha_ml.config.matcha_config import (
+    MatchaConfig,
+    MatchaConfigComponent,
+    MatchaConfigComponentProperty,
+    MatchaConfigService,
+)
+from matcha_ml.errors import MatchaError
+
+
+@pytest.fixture
+def mocked_matcha_config_json_object() -> Dict[str, Dict[str, str]]:
+    """A fixture returning a dictionary representation of the matcha.config.json file.
+
+    Returns:
+        matcha_config (Dict[str, Dict[str, str]]): a dictionary representation of the matcha.config.json file
+    """
+    matcha_config = {
+        "remote_state_bucket": {
+            "account_name": "test_account_name",
+            "container_name": "test_container_name",
+            "resource_group_name": "test_resource_group_name",
+        }
+    }
+
+    return matcha_config
+
+
+@pytest.fixture
+def mocked_matcha_config_component_property() -> MatchaConfigComponentProperty:
+    """A fixture returning a mocked MatchaConfigComponentProperty instance.
+
+    Returns:
+        (MatchaConfigComponentProperty): a mocked MatchaConfigComponentProperty instance.
+    """
+    return MatchaConfigComponentProperty(name="account_name", value="test_account_name")
+
+
+@pytest.fixture
+def mocked_matcha_config_component(
+    mocked_matcha_config_json_object: Dict[str, Dict[str, str]]
+):
+    """A fixture returning a mocked MatchaConfigComponentProperty instance.
+
+    Args:
+        mocked_matcha_config_json_object (Dict[str, Dict[str, str]]): a dictionary representation of the matcha.config.json file
+
+    Returns:
+        (MatchaConfigComponentProperty): a mocked MatchaConfigComponentProperty instance.
+    """
+    properties = []
+    for key, value in mocked_matcha_config_json_object["remote_state_bucket"].items():
+        properties.append(MatchaConfigComponentProperty(name=key, value=value))
+
+    return MatchaConfigComponent(name="remote_state_bucket", properties=properties)
```

Comment:
Nit (alternative): could be a one-liner: `return MatchaConfig(mocked_matcha_config_json_object).components[0]`